component/common/loki/wal/watcher.go

```diff
@@ -212,7 +275,18 @@ func (w *Watcher) watch(segmentNum int) error {
 		// read from open segment routine
 		ok, err := w.readSegment(reader, segmentNum)
 		if debug {
-			level.Warn(w.logger).Log("msg", "Error reading segment inside readTicker", "segment", segmentNum, "read", reader.Offset(), "err", err)
+			level.Warn(w.logger).Log("msg", "Error reading segment inside read ticker or notification", "segment", segmentNum, "read", reader.Offset(), "err", err)
+		}
+
+		// Ignore all errors reading to end of segment whilst replaying the WAL. If error, log a warning accordingly. If not,
+		// return with an ignorable error.
+		if !tail {
+			if err != nil && errors.Unwrap(err) != io.EOF {
+				level.Warn(w.logger).Log("msg", "Ignoring error reading to end of segment, may have dropped data", "segment", segmentNum, "err", err)
+			} else if reader.Offset() != size {
+				level.Warn(w.logger).Log("msg", "Expected to have read whole segment, may have dropped data", "segment", segmentNum, "read", reader.Offset(), "size", size)
+			}
+			return ErrIgnorable
```

Comment:
Aren't ignorable errors called "warnings"? 😄 Is there something special about these ones which makes them more important?

Reply:
You are right actually, we can just log the warnings, and then return nil. Changing to that.